examples/cute/tutorial/xe_gemm.cpp

```diff
@@ -0,0 +1,580 @@
+
+  using WGTile = Shape<_256, _256, _K>;                               // 256x256 WG tile size
+  using SGLayout = Layout<Shape<_8, _4, _1>, Stride<_4, _1, _0>>;     // 8x4 SG tiling, n-major
+
+  using MMA = typename TiledMMAHelper<MMA_Atom<decltype(op)>, Layout<WGTile>, SGLayout>::TiledMMA;
+
+  return MMA{};
+}
+
+template <class ATensor, class BTensor, class CTensor>
+void
+gemm_cute(sycl::queue &Q,
+          ATensor   const& A,         // (M,K)
+          BTensor   const& B,         // (N,K)
+          CTensor        & C)         // (M,N)
+{
+  auto mma = choose_tiled_mma(A, B, C);
+
+  sycl::range<2> local = {size(mma), 1};
+  sycl::range<2> global = {local[0] * ceil_div(shape<0>(B), get<1>(mma.tile_mnk())),
+                           local[1] * ceil_div(shape<0>(A), get<0>(mma.tile_mnk()))};
+
+  namespace syclex = sycl::ext::oneapi::experimental;
+  namespace intelex = sycl::ext::intel::experimental;
+
+  syclex::properties kernel_props {
+    syclex::sub_group_size<16>,
+    intelex::grf_size<256>
+  };
+
+  auto event = Q.parallel_for(sycl::nd_range<2>(global, local), kernel_props,
+    [=](auto) {
+      gemm_device(A, B, C, mma);
+    }
+  );
+
+  EventManager::getInstance().addEvent(event);
+}
+
+template <typename T>
+auto ensure_signed_helper_t() {
+  if constexpr (is_unsigned_v<T>)
+    return make_signed_t<T>{};
+  else
+    return T{};
+}
+
+template <typename T>
+using ensure_signed_t = decltype(ensure_signed_helper_t<T>());
+
+template <class ATensor, class BTensor, class CTensor>
+bool
+gemm_verify(sycl::queue &Q,
+            ATensor const& A,         // (M,K)
+            BTensor const& B,         // (N,K)
+            CTensor const& C)         // (M,N)
+{
+  int m = size<0>(A);
+  int n = size<0>(B);
+  int k = size<1>(A);
+
+  auto ok = sycl::malloc_shared<bool>(1, Q);
+  *ok = true;
+
+  Q.parallel_for(sycl::range<2>(m, n), [=](sycl::item<2> id) {
+    int i = id[0], j = id[1];
+
+    using AccType = typename CTensor::element_type;
+    using SignedAccType = ensure_signed_t<AccType>;
+
+    auto c = AccType(0);
+    for (int h = 0; h < k; h++)
+      c += AccType(A(i,h)) * AccType(B(j,h));
+
+    auto tol = AccType(1e-5f * k);
+    if (std::abs(SignedAccType(c - AccType(C(i,j)))) > tol) {
+#ifdef SHOW_DIFF
+      printf("Error at (%d,%d): got %f, expected %f\n", i, j, double(C(i,j)), double(c));
+#endif
+      *ok = false;
+    }
+  }).wait();
+
+  bool read_ok = *ok;
+
+  sycl::free(ok, Q);
+
+  return read_ok;
+}
+
+template <typename T, char LayoutKind>
+auto
+make_shared_usm_tensor(sycl::queue &Q, int r, int c)
+{
+  auto ptr = make_gmem_ptr(sycl::malloc_shared<T>(r*c, Q));
+  auto shape = make_shape(r, c);
+  if constexpr (LayoutKind == 'C')
+    return make_tensor(ptr, make_layout(shape, make_stride(_1{}, r)));
+  else
+    return make_tensor(ptr, make_layout(shape, make_stride(c, _1{})));
+}
+
+template <typename InTensor>
+void
+free_usm_tensor(InTensor &X, sycl::queue &Q)
+{
+  // RAII? What's that?
```

Comment:
Yeah we really should start using that for our examples. We even already have https://github.com/intel/cutlass-sycl/blob/main/test/unit/common/util.hpp#L81 and are using it for the tests. We should move it to the general purpose util header and start using it for the examples.